util/fileutil/homedir.go

```diff
@@ -0,0 +1,36 @@
+package fileutil
+
+import (
+	"os"
+	"os/user"
+	"runtime"
+)
+
+// HomeDir returns the home directory of the current user, an additional sudoUser
+// is returned if the user is currently running as root
+func HomeDir() (homeDir string, sudoUser *user.User, err error) {
+	if runtime.GOOS == "windows" {
+		homeDir, err := os.UserHomeDir()
+		return homeDir, nil, err
+	}
+	// See if SUDO_USER exists. Use that user's home dir.
+	sudoUserName, ok := os.LookupEnv("SUDO_USER")
```

Comment:
I think @dchw had a nice function for fetching the current user or sudo user in his PR. Might be worth merging them somehow.

Reply:
This code came from the cliutil.DetectHomeDir() function; If we can fold more code into using this function that sounds like a good plan.